### PR TITLE
Ensure CA simulator takes existing nodes into account

### PR DESCRIPTION
**Pull request by vivekbagade**
In findPlaceFor method, the nodes passed aren't taken into account.

### CODE REVIEW

File: cluster-autoscaler/simulator/cluster.go [hunk 1]
```diff
@@ -294,6 +294,10 @@ func findPlaceFor(removedNode string, pods []*apiv1.Pod, nodes map[string]bool,
 		return fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
 	}
 
+	nodeExists := func(nodeName string) bool {
```
Comment (MaciekPytel) [inline cluster-autoscaler/simulator/cluster.go:297]: It's not really what the function is checking - the nodes you're filtering out do exist. Maybe rename it to isCandidateNode or something along those lines?

**Comment by MaciekPytel** [comment]
Thanks for catching that!
Left super small comment, otherwise lgtm.

/approve

File: cluster-autoscaler/simulator/cluster.go (hunk 1 above)
Comment (vivekbagade) [reply cluster-autoscaler/simulator/cluster.go]: Done

**Pull request merged**